web/src/lib/components/asset-viewer/asset-viewer.svelte

```diff
@@ -538,21 +539,24 @@
   $: if (!$user) {
     shouldShowShareModal = false;
   }
-</script>
 
-<svelte:window
-  use:shortcuts={[
+  let shortcutOptions: ShortcutOptions<unknown>[] = [
     { shortcut: { key: 'a', shift: true }, onShortcut: toggleAssetArchive },
     { shortcut: { key: 'ArrowLeft' }, onShortcut: () => navigateAsset('previous') },
     { shortcut: { key: 'ArrowRight' }, onShortcut: () => navigateAsset('next') },
     { shortcut: { key: 'd', shift: true }, onShortcut: () => downloadFile(asset) },
-    { shortcut: { key: 'Delete' }, onShortcut: () => trashOrDelete(false) },
     { shortcut: { key: 'Delete', shift: true }, onShortcut: () => trashOrDelete(true) },
     { shortcut: { key: 'Escape' }, onShortcut: closeViewer },
     { shortcut: { key: 'f' }, onShortcut: toggleFavorite },
     { shortcut: { key: 'i' }, onShortcut: toggleDetailPanel },
-  ]}
-/>
+  ];
+
+  if (!asset.isTrashed) {
```

Comment:
`asset` is reactive, but this statement is not. This has the potential to get out of sync.

Reply:
Understandable. I updated the code. Not sure how I can test it or ensure that it is reactive indeed.

Reply:
Thank you for the changes. Looks all good to me, apart from the `Delete` shortcut not working for trashed assets. I would suggest to keep the shortcuts as they were and only make the following change:

```diff
- { shortcut: { key: 'Delete' }, onShortcut: () => trashOrDelete(false) },
+ { shortcut: { key: 'Delete' }, onShortcut: () => trashOrDelete(asset.isTrashed) },
```

Reply:
Alright! I followed these suggestions. Initially I thought it might be better to disable a shortcut instead of changing the behaviour. I was afraid of accidental deletions because of muscle memory. However, as we have a confirmation dialog, I think it's fine to assign both `Delete` and `Shift+Delete` to "permanently delete" in this situation.